src/vs/base/common/mime.ts

```diff
@@ -225,19 +225,24 @@ export function isUnspecific(mime: string[] | string): boolean {
 	return mime.length === 1 && isUnspecific(mime[0]);
 }
 
+/**
+ * Returns a suggestion for the filename by the following logic:
+ * 1. If a relevant extension exists and is an actual filename extension (starting with a dot), suggest the prefix appended by the first one.
+ * 2. Otherwise, if there are other extensions, suggest the first one.
+ * 3. Otherwise, suggest the prefix.
+ */
 export function suggestFilename(langId: string, prefix: string): string {
-	for (let i = 0; i < registeredAssociations.length; i++) {
-		const association = registeredAssociations[i];
-		if (association.userConfigured) {
-			continue; // only support registered ones
-		}
-
-		if (association.id === langId && association.extension) {
-			return prefix + association.extension;
-		}
+	const extensions = registeredAssociations
```

Comment:
@guywald1 please preserve the `association.userConfigured` semantics

Reply:
Ah, wanted to add it to add it the `.filter` but it slipped my mind. Will do :)